app/src/main/java/com/iron/espresso/presentation/home/study/list/StudyListViewModel.kt

```diff
@@ -7,21 +7,72 @@ import com.iron.espresso.AuthHolder
 import com.iron.espresso.Logger
 import com.iron.espresso.base.BaseViewModel
 import com.iron.espresso.data.model.StudyItem
+import com.iron.espresso.di.ApiModule
+import com.iron.espresso.domain.repo.StudyRepository
 import com.iron.espresso.ext.networkSchedulers
 import com.iron.espresso.ext.plusAssign
 import com.iron.espresso.ext.toErrorResponse
-import com.iron.espresso.model.api.StudyApi
 import retrofit2.HttpException
+import java.util.concurrent.TimeUnit
 
-class StudyListViewModel @ViewModelInject constructor(private val studyApi: StudyApi) :
+class StudyListViewModel @ViewModelInject constructor(private val studyRepository: StudyRepository) :
     BaseViewModel() {
 
     private val _studyList = MutableLiveData<List<StudyItem>>()
     val studyList: LiveData<List<StudyItem>>
         get() = _studyList
 
+    private val _scrollItem = MutableLiveData<List<StudyItem>>()
+    val scrollItem: LiveData<List<StudyItem>>
+        get() = _scrollItem
+
+    private val allList = mutableListOf<StudyItem>()
+    private var isPaging = false
+
+    private fun firstItemResult(studyList: List<StudyItem>): MutableList<StudyItem> {
+        val list: MutableList<StudyItem> = mutableListOf()
+        allList.clear()
+        allList.addAll(studyList)
+
+        studyList.forEach { studyItem ->
+            if (studyItem.isPaging) {
+                isPaging = true
+                return@forEach
+            } else {
+                list.add(studyItem)
+            }
+        }
+        return list
+    }
+
+    private fun scrollMoreItem(startSize: Int): String {
+        val scrollIdList = mutableListOf<Int>()
+        val pagingSize = studyList.value?.size ?: 0
+        val endSize = startSize + pagingSize
+        return when {
+            endSize <= allList.size -> {
+                for (i in startSize until endSize) {
+                    scrollIdList.add(allList[i].id)
+                }
+                scrollIdList.joinToString(",")
+
+            }
+            endSize > allList.size -> {
+                isPaging = false
+                for (i in startSize until allList.size) {
+                    scrollIdList.add(allList[i].id)
+                }
+                scrollIdList.joinToString(",")
+            }
+            else -> {
+                isPaging = false
+                error("validation error")
+            }
+        }
+    }
+
     fun getStudyList(category: String, sort: String) {
-        compositeDisposable += studyApi
+        compositeDisposable += ApiModule.provideStudyApi()
```

Comment:
```suggestion
        compositeDisposable += studyRepository
```
이렇게 사용가능하도록 함수를 구성하는게 죠을듯

Reply:
페이징이 아닌 스터디 목록 또한 적용시키겠습니다!

Reply:
9cda892 @wswon 수정하였습니다